File: min_max_sum.py
# https://www.hackerrank.com/challenges/mini-max-sum/problem

def miniMaxSum(arr):
    def tap(x, f):
        f(x)
        return x

    arr.sort()
    without_min_sum = sum(tap(arr, lambda x: x.pop(0)))
    without_max_sum = sum(tap(arr, lambda x: x.pop()))
    return [without_max_sum, without_min_sum]


# print(miniMaxSum([1, 2, 3, 4, 5]))

File: min_max_sum.rb
# https://www.hackerrank.com/challenges/mini-max-sum/problem

def miniMaxSum(arr)
    sorted_arr = arr.sort
    without_min_sum = sorted_arr.drop(1).sum
    without_max_sum = sorted_arr.tap(&:pop).sum
    [without_max_sum, without_min_sum]
end

# miniMaxSum([7, 69, 2, 221, 8974])
